lib/reqIdGenFactory.js

```diff
@@ -1,16 +1,25 @@
 'use strict'
 
+function patchNumberLookup (numberLookup) {
+  for (let i = 0; i < 36; ++i) {
+    numberLookup[i] = '0' + numberLookup[i]
+  }
+}
+
 module.exports = function (requestIdHeader, optGenReqId) {
-  // 2,147,483,647 (2^31 − 1) stands for max SMI value (an internal optimization of V8).
-  // With this upper bound, if you'll be generating 1k ids/sec, you're going to hit it in ~25 days.
-  // This is very likely to happen in real-world applications, hence the limit is enforced.
-  // Growing beyond this value will make the id generation slower and cause a deopt.
-  // In the worst cases, it will become a float, losing accuracy.
-  const maxInt = 2147483647
-  let nextReqId = 0
+  const twoDigitPermutationCount = 36 ** 2
+  const numberLookup = new Array(twoDigitPermutationCount).fill(0).map((v, i) => i.toString(36))
+  let reqMajorCounter = 0
+  let reqSubCounter = 1
+  let prefix = 'req-'
+
   function defaultGenReqId (req) {
-    nextReqId = (nextReqId + 1) & maxInt
-    return `req-${nextReqId.toString(36)}`
+    if (reqSubCounter === twoDigitPermutationCount) {
+      reqSubCounter = 0
+      reqMajorCounter === 0 && patchNumberLookup(numberLookup)
```

Comment:
Couldn't it be evaluated in startup time? `reqMajorCounter = patchNumber(numberLookup)`

Reply:
No. 

numberLookup are all the numbers from 0 to 1296 on base 36. The first 36 numbers are single digits, so 0 => '0', 1 => '1' and so on. After we increase reqMajorCounter the first time, we need double digits in the numberLookup. So we patch those to be double digits, so 0 => '00', 1 => '01'  and so on.